db/db.go

```diff
@@ -58,11 +59,15 @@ type db struct {
 
 	events events.Events
 
-	parser core.Parser
+	parser       core.Parser
+	lensRegistry client.LensRegistry
 
 	// The maximum number of retries per transaction.
 	maxTxnRetries immutable.Option[int]
 
+	// The maximum number of cached migrations instances to preserve per schema version.
+	lensPoolSize immutable.Option[int]
```

Comment:
This is a pool of identical migration instances for a single schema version.  The instances are not thread safe, and can only be used by one context at a time.  If there was only one instance it would essentially force a large chunk of fetcher-effort through a single thread.